collect_app/src/test/java/org/odk/collect/android/storage/migration/StorageMigratorTest.java

```diff
@@ -0,0 +1,163 @@
+package org.odk.collect.android.storage.migration;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.odk.collect.android.storage.StoragePathProvider;
+import org.odk.collect.android.storage.StorageStateProvider;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class StorageMigratorTest {
+
+    private StorageMigrator storageMigrator;
+    private StorageEraser storageEraser;
+    private final StoragePathProvider storagePathProvider = spy(StoragePathProvider.class);
+    private final StorageStateProvider storageStateProvider = mock(StorageStateProvider.class);
+
+    @Before
+    public void setup() {
+        storageEraser = spy(new StorageEraser(storagePathProvider));
+        storageMigrator = spy(new StorageMigrator(storagePathProvider, storageStateProvider, storageEraser));
+
+        doNothing().when(storageMigrator).reopenDatabases();
```

Comment:
I think we need to break this object up a bit so we're not having to fake/mock methods on the object under test. It looks like you've had to do this because of a bunch of static methods though. Maybe using a similar strategy to the `BackgroundLocationHelper` that @lognaturel created? This worked at the time as it allowed her to wrap a bunch of static methods/singletons that the new object she was build needed without having to go and refactor/unwrap all of them. This interface could also include `isFormUploaderRunning()` and `isFormDownloaderRunning()`.

Reply:
Agreed that sounds like a promising direction.